### PR TITLE
Test isfinite

### PR DESCRIPTION
**Pull request by dneto0**
<hr>

**Author checklist for test code/plans:**

- [x] All outstanding work is tracked with "TODO" in a test/file description or `.unimplemented()` on a test.
- [x] New helpers, if any, are documented using `/** doc comments */` and can be found via `helper_index.txt`.
- [x] (Optional, sometimes not possible.) Tests pass (or partially pass without unexpected issues) in an implementation. (Add any extra details above.)

**[Reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md) for test code/plans:** (Note: feel free to pull in other reviewers at any time for any reason.)

- [ ] The test path is reasonable, the [description](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) "describes the test, succinctly, but in enough detail that a reader can read only the test plans in a file or directory and evaluate the completeness of the test coverage."
- [ ] Tests appear to cover this area completely, except for outstanding TODOs. Validation tests use control cases.
    (This is critical for coverage. Assume anything without a TODO will be forgotten about forever.)
- [ ] Existing (or new) test helpers are used where they would reduce complexity.
- [ ] TypeScript code is readable and understandable (is unobtrusive, has reasonable type-safety/verbosity/dynamicity).

### CODE REVIEW

File: src/webgpu/shader/execution/builtin/isinf.spec.ts [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+      };
+    };
+    const cases = [
+      // Non-infinity
+      make(0.0, false),
+      make(10.0, false),
+      make(-10.0, false),
+      // Infinities
+      make(Infinity, true),
+      make(-Infinity, true),
```
Comment (alan-baker) [inline src/webgpu/shader/execution/builtin/isinf.spec.ts:48]: The WGSL spec says:
> Implementations may assume that NaNs, infinities are not present

That implies isInf(INF) might return false. Similar issue for `isNan`.

File: src/webgpu/shader/values.ts [hunk 2]
```diff
@@ -63,3 +63,29 @@ export function normalF32Examples(): Array<number> {
...
+  const sign_bits: [number, number] = [0, 0x8000_0000];
+  for (const sign_bit in sign_bits) {
+    const sign_and_exponent = ((sign_bit as unknown) as number) | exponent_bit;
+    const bits = sign_and_exponent | 0x40_0000;
+    // Only the most significant bit of the mantissa is set.
+    result.push(bits);
+
+    // Quiet and signalling NaNs differ based on the most significant bit
+    // of the mantissa. Try both.
+    for (const quiet_signalling in [0, 0x40_0000]) {
```
Comment (ben-clayton) [inline src/webgpu/shader/values.ts:82]: See review comments in https://github.com/gpuweb/cts/pull/800#pullrequestreview-798045365

File: src/webgpu/shader/execution/builtin/isinf.spec.ts (hunk 1 above)
Comment (dneto0) [reply src/webgpu/shader/execution/builtin/isinf.spec.ts]: outdated.

File: src/webgpu/shader/values.ts (hunk 2 above)
Comment (dneto0) [reply src/webgpu/shader/values.ts]: fixed in the base commit.

**Comment by dneto0** [comment]
isFinite has been removed from WGSL
https://github.com/gpuweb/gpuweb/pull/2325

**Pull request closed without merging**